packages/lib/src/vis/hooks.ts

```diff
@@ -104,3 +104,36 @@ export function useCustomColors(
 
   return [colors, refCallback];
 }
+
+export function useRaycasterThreshold(
+  lineOrPoints: 'Line' | 'Points',
+  desiredThreshold: number
+) {
+  const raycaster = useThree((state) => state.raycaster);
+  const zoomVector = useThree((state) => state.camera.scale);
+
+  useEffect(() => {
+    const params = raycaster.params[lineOrPoints];
+    const oldThreshold = params ? params.threshold : 1;
+
+    if (params) {
+      params.threshold = desiredThreshold;
+    }
+
+    return () => {
+      if (params) {
+        params.threshold = oldThreshold;
+      }
+    };
+  }, [raycaster, lineOrPoints, desiredThreshold]);
+
+  useFrame(() => {
+    // Reduce the threshold dynamically when zooming as the size of Line and Points objects is reduced in world units
+    const zoom = zoomVector.x;
```

Comment:
Even though this is fine, since the `zoomVector` is mutated by Three, I would prefer if you used `camera.scale` here and if your retrieved the full `camera` object from `useThree`. This is more consistent with what we do in other places.